### DISCUSSION TITLE
Postgres connections used up after installing user-migration app

### DISCUSSION THREAD

**Discussion by shaib** [nextcloud/all-in-one#3525] [Questions] [answered]
Hi,

Continuing the adventure from #3524 -- I installed the user-migration app, and the server immediately died ("Intenal server error"). When I looked at its logs, it was clear that the issue is that all the Postgresql connections were busy, just like in #2839 -- except that the solution from there had already been applied, I had `'dbpersistent' => true, ` "out of the box".

I restarted the DB, and I also had to restart the nextcloud container, and everything came back. I was then able to import some user data, and also install another app, with no incident.

Is there any more info I can provide which would be useful to prevent this? Or should it be reported strictly to the app authors?

**Comment by szaimen** [comment]
Hi, can you upgrade to AIO v7.4.1 Beta and check if it resolves this? See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by shaib** [comment]
I now have a working instance which I need to keep up for a few days, I may be able to try next week.
However, I haven't try to reproduce the problem even on the current version...

**Reply by shaib** [reply]
Hmm. So, yesterday I was expecting a couple of users to join the server, and they only really joined today. A little after they joined, the PG connections were all busy again. This time, there was no administrative operation involved (well, except that it really was their first login, and the demo files were created in their home folders).

According to both the user reports and the logs (apache log; the nextcloud log, AFAICT, holds nothing before the "can't connect, PG busy" errors) -- the only "interesting" thing done was that a user created a Calendar event, inviting two others (one of them is me; I received the invitation mail, but I see no event in my calendar, even having accepted it -- although this may be related to the fact that I only looked at it after the event has ended).

Will logs be helpful?  At least to help reproduce?

**Comment by shaib** [comment, marked as answer]
Changed to beta. On the way there, as reported in #3524, I removed a `daemon.json` file which pointed to remote DNS servers. I already see better behavior -- when looking at the open connections (`select * from pg_stat_activity`), before, every operation on the site kept a few connections open for a few tens of seconds. This is no longer the case. I hope it stays well.

Are you aware of specific relevant changes in the code between `latest` and `beta`?

**Reply by szaimen** [reply]
> Are you aware of specific relevant changes in the code between `latest` and `beta`?



Yes, see https://github.com/nextcloud/all-in-one/pull/3479
